src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/CostFunctionsForPhysicalPlansImpl.java

```diff
@@ -0,0 +1,117 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.PhysicalPlan;
+import se.liu.ida.hefquin.engine.queryproc.QueryOptimizationException;
+import se.liu.ida.hefquin.engine.queryproc.impl.optimizer.utils.CardinalityEstimation;
+
+public class CostFunctionsForPhysicalPlansImpl implements CostFunctionsForPhysicalPlans{
+    protected final CostFunctionsForRootOperatorsImpl costFunctionForRoot;
+
+    public CostFunctionsForPhysicalPlansImpl(CardinalityEstimation cardEstimate) {
+        this.costFunctionForRoot = new CostFunctionsForRootOperatorsImpl(cardEstimate);
+    }
+
+    public CostOfPhysicalPlan getCostOfPhysicalPlan( final PhysicalPlan pp ) throws QueryOptimizationException {
```

Comment:
Add this method also to the interface.

After doing so, add the annotation `@Override`